biz.aQute.bndlib/src/aQute/bnd/build/Project.java

```diff
@@ -1797,16 +1797,93 @@ public boolean isNoBundles() {
 
 	public File saveBuild(Jar jar) throws Exception {
 		try {
-			File f = getOutputFile(jar.getName(), jar.getVersion());
+			File outputFile = getOutputFile(jar.getName(), jar.getVersion());
+			File logicalFile = outputFile;
+
 			String msg = "";
-			if (!f.exists() || f.lastModified() < jar.lastModified()) {
-				reportNewer(f.lastModified(), jar);
-				IO.deleteWithException(f);
-				File fp = f.getParentFile();
+			if (!outputFile.exists() || outputFile.lastModified() < jar.lastModified()) {
+				reportNewer(outputFile.lastModified(), jar);
+
+				File fp = outputFile.getParentFile();
 				if (!fp.isDirectory()) {
 					IO.mkdirs(fp);
 				}
-				jar.write(f);
+
+				// On windows we sometimes cannot delete a file because
+				// someone holds a lock in our or another process. So if
+				// we set the -overwritestrategy flag we use an avoiding
+				// strategy.
+				// We will always write to a temp file name. Basically the
+				// calculated name + a variable suffix. We then create
+				// a link with the constant name to this variable name.
+				// This allows us to pick a different suffix when we cannot
+				// delete the file. Yuck, but better than the alternative.
+
+				String overwritestrategy = getProperty("-overwritestrategy", "classic");
+				swtch: switch (overwritestrategy) {
+					case "delay" :
```

Comment:
So this delay happens synchronously in the incremental build process? I guess I'm okay with this but it needs to be documented very clearly that the delay option can add up to 5s to the build of every bundle.